accepted/future-releases/extension-types/feature-specification.md

```diff
@@ -601,6 +605,28 @@ If _Dm_ is a method with function type `F`, and `args` exists, the static
 analysis of the extension type member invocation is the same as that of an
 invocation with argument part `args` of a function with the given type.
 
+### Special-case Typing Rules for `num` Members
+
+The language specification contains special typing rules for invocations
+of `+`, `-`, `*`, `%`, `remainder` and `clamp` methods on 
+subtypes of `num`, `int` and `double`.
+
+Those rules do *not* apply to invocations of extension or extension type
+members.
+The rules are otherwise unchanged when applied to non-extension type
+member invocations, which will necessarily invoke the corresponding members
+of an instance of `int` or `double`.
+
+_It's not currently possible to have a subtype of `num` which has
+an extension member for those member names, but if it ever becomes possible,
```

Comment:
```dart
extension E on int {
  String operator +(String other) => '$other, ';
  String get g => '${this + 'Hello'}world!';
}

void main() {
  print(42.g);
}
```

Seems that it should be possible to have exactly one of those extension members, but it would not be possible to invoke it implicitly except in the body of the extension declaration. It looks like the explicitly resolved extension member invocation `E(42) + 'Hello'` _does_ get the correct treatment, but the current behavior of both the CFE and the analyzer is to report an error for the implicitly resolved case.

Reply:
Sorry, that's not true: `this + 'Hello'` does _not_ perform a lexical lookup for operator `+`, it is an invocation with an explicit receiver (`this`, in an extension member body, which means that it will be subject to normal resolution and run the instance operator).

Closing.